### ISSUE TITLE
[🐛 Bug]: Selenium 4.15 with C# throwing error - Unable to obtain chrome using Selenium Manager when i am running tests in parallel mode 

### ISSUE THREAD

**Issue by pravinshinde222** [SeleniumHQ/selenium#13157]
### What happened?

I am using selenium 4.15.0 version having selenium manager with C# , and running my tests in parallel mode using azure DevOps,
But i am getting below exception 

Test method <TestMethodName> threw exception:
System.InvalidOperationException: Unable to start Chrome. Killed chromedriver process! Unable to obtain chrome using Selenium Manager; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location
TestCleanup method <TestMethodName>.TestCleanUp threw exception. System.NullReferenceException: System.NullReferenceException: The WebDriver browser instance was not initialized. You should first call the method Start..

Earlier i was using 4.8.1 version of selenium which was working fine with webdrivermanager on parallel execution, but when we started using 4.15, we are getting above error 

### How can we reproduce the issue?

```shell
Please find below sample code which is throwing exception

public class SeleniumHelper
    {
        public IWebDriver driver;
        public void initDriver()
        {
            ChromeDriverService chromeDriverService = null;
            chromeDriverService = ChromeDriverService.CreateDefaultService();
            chromeDriverService.HideCommandPromptWindow = true;
            ChromeOptions options = new ChromeOptions();
            options.AddArgument("--disable-infobars");
            options.AddArgument("--start-maximized");
            try
            {
                driver = new ChromeDriver(chromeDriverService, options);
            }
            catch (Exception e)
            {
                var chromeProcesses = Process.GetProcessesByName("chromedriver");
                foreach (var chromeProcess in chromeProcesses)
                {
                    if (chromeProcess.Id == chromeDriverService.ProcessId)
                    {
                        chromeProcess.Kill();
                    }
                }
                throw new InvalidOperationException("Unable to start Chrome. Killed chromedriver process! " + e.Message);
            }
        }
    }
```

### Relevant log output

```shell
.NET does not currently have a Logging implementation
```

### Operating System

Windows 10 enterprises

### Selenium version

4.15

### What are the browser(s) and version(s) where you see this issue?

Chrome version 117

### What are the browser driver(s) and version(s) where you see this issue?

As tests are running on agent machine, so unable to check

### Are you using Selenium Grid?

Not using selenium grid

**Comment by titusfortner** [comment]
First, your code is killing all chromedriver processes when one execution has an exception. This is very bad if you are trying to run in parallel.

The code itself looks correct, but we need to see the full stack trace and have some way to reproduce the issue.

**Comment by nvborisenko** [comment]
@pravinshinde222 please change your code from
```csharp
throw new InvalidOperationException("Unable to start Chrome. Killed chromedriver process! " + e.Message);
```

to

```csharp
throw new InvalidOperationException("Unable to start Chrome. Killed chromedriver process! " + e.ToString());
```

It will allow us to see the exact issue behind.

**Comment by dcdraper** [comment]
I am seeing what seems to be a similar issue in 4.15 when running tests in parallel. We are using a .runsettings file and setting NumberOfTestWorkers to 8.  Only about three tests fail at the beginning of each run.  It seems that once chrome is installed by one of the early tests/threads that the issue no longer presents.
Version 4.15 is the first version we have tried using Selenium Manager,

Please let me know if you prefer a separate issue report for this one. 

.runsettings file:
`<?xml version="1.0" encoding="utf-8" ?>
<RunSettings>
	
	<TestRunParameters>
		<Parameter name="environment" value="DEV" />
		<Parameter name="browser" value="CH_HEADLESS_MODE" />
		<Parameter name="gridExecution" value="false" />
		<Parameter name="includeScreenshotsInReport" value="true" />
		<Parameter name="generateFileWithFailedTests" value="true" />
		<Parameter name="loglevel" value="INFO" />
	</TestRunParameters>
	<NUnit>
		<NumberOfTestWorkers>8</NumberOfTestWorkers>
	</NUnit>
</RunSettings>`

From the execution log:
OpenQA.Selenium.NoSuchDriverException : Unable to obtain chrome using Selenium Manager; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors/driver_location
----> OpenQA.Selenium.WebDriverException : Error starting process: C:\a\1\s\SanityTestPack\bin\Debug\net7.0\selenium-manager\windows\selenium-manager.exe  --browser "chrome" --output json
----> OpenQA.Selenium.WebDriverException : Selenium Manager process exited abnormally with 65 code: C:\a\1\s\SanityTestPack\bin\Debug\net7.0\selenium-manager\windows\selenium-manager.exe  --browser "chrome" --output json
Standard Output >>
{
"logs": [
{
"level": "ERROR",
"timestamp": 1702678423,
"message": "The process cannot access the file because it is being used by another process. (os error 32)"
}
],
"result": {
"code": 65,
"message": "The process cannot access the file because it is being used by another process. (os error 32)",
"driver_path": "",
"browser_path": ""
}
}

<<

Stack trace
at OpenQA.Selenium.DriverFinder.FullPath(DriverOptions options)
at OpenQA.Selenium.Chromium.ChromiumDriver.GenerateDriverServiceCommandExecutor(DriverService service, DriverOptions options, TimeSpan commandTimeout)
at OpenQA.Selenium.Chromium.ChromiumDriver..ctor(ChromiumDriverService service, ChromiumOptions options, TimeSpan commandTimeout)
at OpenQA.Selenium.Chrome.ChromeDriver..ctor(ChromeDriverService service, ChromeOptions options, TimeSpan commandTimeout)
at OpenQA.Selenium.Chrome.ChromeDriver..ctor(ChromeOptions options)
[at SanityTestPack.TestBase.Base.StartWebBrowser() in C:\a\1\s\SanityTestPack\TestBase\Base.cs:line 278](https://dev.azure.com/GoEkos/Test%20Automation%20Framework/_git/05457c59-b0fd-421c-8963-dc5e0e85d9cf?path=%2FSanityTestPack%2FTestBase%2FBase.cs&version=GBworkitem%2F9791%2Fuse-selenium-manager&_a=contents&line=278&lineEnd=279&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
[at SanityTestPack.TestBase.Base.InitializeBrowser(String url, String browser) in C:\a\1\s\SanityTestPack\TestBase\Base.cs:line 114](https://dev.azure.com/GoEkos/Test%20Automation%20Framework/_git/05457c59-b0fd-421c-8963-dc5e0e85d9cf?path=%2FSanityTestPack%2FTestBase%2FBase.cs&version=GBworkitem%2F9791%2Fuse-selenium-manager&_a=contents&line=114&lineEnd=115&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
[at SanityTestPack.TestBase.Base.BeforeTestScenario() in C:\a\1\s\SanityTestPack\TestBase\Base.cs:line 41](https://dev.azure.com/GoEkos/Test%20Automation%20Framework/_git/05457c59-b0fd-421c-8963-dc5e0e85d9cf?path=%2FSanityTestPack%2FTestBase%2FBase.cs&version=GBworkitem%2F9791%2Fuse-selenium-manager&_a=contents&line=41&lineEnd=42&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
at InvokeStub_Action`1.Invoke(Object, Object, IntPtr*)
at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.FireEvents(HookType hookType)
at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.FireScenarioEvents(HookType bindingEvent)
at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.OnScenarioStart()
at TechTalk.SpecFlow.Infrastructure.TestExecutionEngine.OnAfterLastStep()
at TechTalk.SpecFlow.TestRunner.CollectScenarioErrors()
at SanityTestPack.Specs.ActivityTypeFeature.ScenarioCleanup()
[at SanityTestPack.Specs.ActivityTypeFeature.Activity_Type_Create_Edit_Delete() in C:\a\1\s\SanityTestPack\Specs\ActivityType.feature:line 21](https://dev.azure.com/GoEkos/Test%20Automation%20Framework/_git/05457c59-b0fd-421c-8963-dc5e0e85d9cf?path=%2FSanityTestPack%2FSpecs%2FActivityType.feature&version=GBworkitem%2F9791%2Fuse-selenium-manager&_a=contents&line=21&lineEnd=22&lineStartColumn=1&lineEndColumn=1&lineStyle=plain)
at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
--WebDriverException
at OpenQA.Selenium.SeleniumManager.RunCommand(String fileName, String arguments)
at OpenQA.Selenium.SeleniumManager.DriverPath(DriverOptions options)
at OpenQA.Selenium.DriverFinder.FullPath(DriverOptions options)
--WebDriverException
at OpenQA.Selenium.SeleniumManager.RunCommand(String fileName, String arguments)

**Comment by nvborisenko** [comment]
I suppose it would be a separate issue, please post new one with details. BTW I wasn't success to reproduce it.

**Comment by dcdraper** [comment]
Well I cannot reproduce it either now.  If it happens again I will write a new issue for it.

**Comment by titusfortner** [comment]
Closing because we didn't get more info here. We can open a new ticket if there's still an issue in 4.16
